### ISSUE TITLE
Ref calls not printed with `--print_ref_calls` when `--vcf_fn=FILE` is specified

### ISSUE THREAD

**Issue by gtollefson** [HKU-BAL/Clair3#261]
Hi,

When I run the following clair3 command with --vcf_fn and --print_ref_calls provided, reference records report the genotype as ./. and don't report the other format fields (DP, etc.)

```
singularity exec \
			-B {params.INPUT_DIR} \
		  	-B {params.OUTPUT_DIR} \
		  	-B {params.REF_DIR} \
			-B {params.RESOURCE_DIR} \
		  	{input.sif_path} \
		  	/opt/bin/run_clair3.sh \
		  	--bam_fn={input.samples_to_run} \
		  	--ref_fn={input.REF} \
		  	--model_path=/opt/models/{params.MODEL_NAME} \
		  	--output={params.OUTPUT_DIR} \
		  	--threads={params.THREADS} \
		  	--platform=ont \
		  	--include_all_ctgs \
		  	--no_phasing_for_fa \
		  	--sample_name={wildcards.samples} \
		  	--gvcf \
		  	--include_all_ctgs \
			--threads=1 \
			--vcf_fn={params.targets_vcf} \
			--print_ref_calls

```
VCF output example showing missing ref fields:
<img width="1263" alt="Screenshot 2024-01-10 at 11 48 37 PM" src="https://github.com/HKU-BAL/Clair3/assets/32968172/f924943e-7734-455f-9461-5a3fa48f0488">

When I run a the same clair3 command but without --vcf_fn, the reference genotypes are reported as 0/0 and other format fields are reported.

```
singularity exec \
			-B {params.INPUT_DIR} \
		  	-B {params.OUTPUT_DIR} \
		  	-B {params.REF_DIR} \
			-B {params.RESOURCE_DIR} \
		  	{input.sif_path} \
		  	/opt/bin/run_clair3.sh \
		  	--bam_fn={input.samples_to_run} \
		  	--ref_fn={input.REF} \
		  	--model_path=/opt/models/{params.MODEL_NAME} \
		  	--output={params.OUTPUT_DIR} \
		  	--threads={params.THREADS} \
		  	--platform=ont \
		  	--include_all_ctgs \
		  	--no_phasing_for_fa \
		  	--sample_name={wildcards.samples} \
		  	--gvcf \
		  	--include_all_ctgs \
			--threads=1 \
			--print_ref_calls
```
VCF output example showing present ref fields:

<img width="1221" alt="Screenshot 2024-01-10 at 11 49 51 PM" src="https://github.com/HKU-BAL/Clair3/assets/32968172/8734d738-48a0-4235-9f92-cc497b454f90">

Is this expected behavior? I'd like to be able to run clair3 to report full records for reference calls with GT, GQ, and DP values when --vcf_fn is specified.

**Comment by aquaskyline** [comment]
Yes it is expected. In the second example, you enabled --gvcf, which incurs more calculations at the reference calls, thus giving you more details.

**Comment by gtollefson** [comment]
@aquaskyline Thanks for your response, but I don't think that answers my question since I enabled `--gvcf` for both examples - not just the second example. `--vcf_fn` is the only difference between the two runs.

Also the output examples I pasted above are both VCF file output (not gvcf files) from runs with --gvcf enabled which are automatically written along with the gvcf files. The gvcf output from the same run with --gvcf and --vcf_fn enabled does show 0/0 calls. I've pasted a screenshot from this run showing that the gvcf output shows 0/0 calls on the same sites that are reported as ./. in the vcf output. 

Ref genotype missing at: chr4 748134 in VCF
![Screenshot 2024-01-16 at 12 00 39 PM](https://github.com/HKU-BAL/Clair3/assets/32968172/ee9d165a-a533-4ea9-8065-754bd1710080)

Ref genotype present at: chr4 748134 in GVCF from same run:
![Screenshot 2024-01-16 at 12 00 49 PM](https://github.com/HKU-BAL/Clair3/assets/32968172/fc3042a9-2fdb-4f40-9efb-afecf13a7186)

This suggests that 0/0 genotypes and associated DP, GQ, AD values aren't reported in the **VCF output** (but are reported in the gvcf output) when `--vcf_fn` is specified. 

We want to use the VCF output in our pipelines, but also need to run clair3 with '--vcf_fn' while outputting 0/0 reference genotypes and associated fields (DP, GQ, AD etc). Is this possible?

**Comment by aquaskyline** [comment]
Do you mean you were using `--vcf_fn` to do genotyping only at the sites you are interested in, but you also expect DP, GQ, AD to be printed, which is currently lacking in Clair3's `--vcf_fn` mode? I need some clarifications here cuz you also mentioned `--gvcf`, but `--gvcf` gives block output at reference calls, thus no per site DP and AD.

**Comment by zhengzhenxian** [comment]
Hi, @gtollefson,

The './.' variants indicate low-AF sites that were not detected by our variant caller and were subsequently included during postprocessing. To output these variants as `RefCall`, you can try adding the `--snp_min_af=0.0` and `--indel_min_af=0.0` options. Additionally, we will set these two parameters as default in the '--vcf_fn' mode.

And please note that the GVCF output of the given site is calculated in the preprocessing step, and may not directly align with the fields in the VCF output.

**Comment by aquaskyline** [comment]
@gtollefson may we collect your feedback on whether the two settings --snp_min_af=0.0 and --indel_min_af=0.0 solve your problem?

**Comment by aquaskyline** [comment]
--snp_min_af=0.0 and --indel_min_af=0.0 actually refers to variant AF, a site with reference AF 0.9094 is with variant AF <0.1

**Comment by gtollefson** [comment]
Ok I see! Thank you. We appreciate your fast response and clean solution. 
Best,
George

**Comment by aquaskyline** [comment]
Fixed in v1.0.6.

**Comment by bioinfoAG** [comment]
I was wondering, how could I make Clair3 work with the above-mentioned flags (vcf_fn; --snp_min_af=0.0; --indel_min_af=0.0; --print_ref_calls) for indels as well? Because now it seems to be working well for genotyping set positions, but it doesn't expand to account for indels.
